File: src/main/java/top/htext/botools/Botools.java
package top.htext.botools;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import top.htext.botools.commands.BotoolsCommand;

public class Botools implements ModInitializer {
    public static final Logger LOGGER = LogManager.getLogger(Botools.MOD_NAME);
    public static final String MOD_NAME = "Botools";

    @Override
    public void onInitialize() {
        LOGGER.info("Botools initialized.");

        CommandRegistrationCallback.EVENT.register(((dispatcher, registryAccess, environment) -> BotoolsCommand.register(dispatcher)));
    }
}

File: src/main/kotlin/top/htext/botools/Botools.kt
package top.htext.botools

import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Botools: ModInitializer {
	
	override fun onInitialize() {
		logInfo("Mod load Initiated.")
		
		CommandRegistrationCallback.EVENT.register {dispatcher,dedicated->
			BotCommandRegister().register(dispatcher)
		}
	}
}

private val logger: Logger = LoggerFactory.getLogger("Botools")

fun logInfo(message: String){ logger.info(message) }